plugin/reload/reload.go

```diff
@@ -54,7 +56,9 @@ func hook(event caddy.EventName, info interface{}) error {
 					md5sum = s
 					// now lets consider that plugin will not be reload, unless appear in next config file
 					// change status iof usage will be reset in setup if the plugin appears in config file
+					mu.Lock()
 					r.usage = maybeUsed
+					mu.Unlock()
 					_, err := instance.Restart(corefile)
 					if err != nil {
 						log.Printf("[ERROR] Corefile changed but reload failed: %s\n", err)
```

Comment:
About the usage of the Mutex.
It is here a very usual issue of sharing the same variable based on struct.

I think would be much proper if you define Getter and Setter on this global variable with mutex inisde these Setter / Getter.
You can group the getting ans setting of the properties as needed.

and add a big comment : do not access out of these Getter/Setter.